op/model.py

```diff
@@ -313,4 +323,14 @@ def config_get(self):
         return self._run('config-get')
 
     def is_leader(self):
-        return self._run('is-leader')
+        """Obtain the current leadership status for the unit the charm code is executing on.
+
+        The value is cached for the duration of a lease which is 30s in Juju.
+        """
+        if datetime.timedelta(seconds=time.monotonic() - self._leader_check_time) > self.LEASE_REFRESH_PERIOD or self._is_leader is None:
```

Comment:
as a minor optimization, and as a safety measure, you probably want to do:

now = time.monotonic()
time_since_check = datetime.timedelta(second=now - self._leader_check_time)
if (self._is_leader is None or time_since_check > self.LEASE_REFRESH_PERIOD):
    self._is_leader = self._run('is-leader')
    self._leader_check_time = now
return self._is_leader

That avoids calling time.monotonic() twice and IMO is a bit easier to read. The comment still holds, since we call time.monotonic() before the IsLeader call.

Reply:
Agreed, thanks.